tests/inc/classes/test-class-blank-theme.php

```diff
@@ -0,0 +1,200 @@
+<?php
+/**
+ * Test_Blank_Theme class for base theme class of the theme..
+ *
+ * @author  Kiran Potphode <kiran.potphode@rtcamp.com>
+ *
+ * @package Blank_Theme
+ */
+
+namespace BLANK_THEME\Tests;
+
+use Exception;
+use BLANK_THEME\Inc\BLANK_THEME;
+
+/**
+ * Class Test_Blank_Theme
+ *
+ * @coversDefaultClass \BLANK_THEME\Inc\BLANK_THEME
+ */
+class Test_Blank_Theme extends \WP_UnitTestCase {
+	/**
+	 * This Assets data member will contain Assets class object.
+	 *
+	 * @var \BLANK_THEME\Inc\BLANK_THEME
+	 */
+	protected $instance = false;
+
+	/**
+	 * This function activate the theme.
+	 *
+	 * @return void
+	 */
+	public function setUp() : void {
+
+		parent::setUp();
+		switch_theme( 'blank-theme' );
+		$this->instance  = BLANK_THEME::get_instance();
+		$this->mock_post = $this->factory()->post->create_and_get();
+		$GLOBALS['post'] = $this->mock_post;
+		add_filter( 'is_active_sidebar', array( $this, 'deactivate_sidebar' ), 10, 2 );
+	}
+
+	public function tearDown() {
+		unset( $GLOBALS['post'] );
+	}
+
+	/**
+	 * Test constructor function.
+	 *
+	 * @covers ::__construct
+	 * @covers ::_setup_hooks
+	 */
+	public function test_construct() {
+
+		Utility::invoke_method( $this->instance, '__construct' );
+		$this->assertInstanceOf( 'BLANK_THEME\Inc\BLANK_THEME', $this->instance );
+		$hooks = array(
+			array(
+				'type'     => 'filter',
+				'name'     => 'excerpt_more',
+				'priority' => 10,
+				'function' => 'add_read_more_link',
+			),
+			array(
+				'type'     => 'filter',
+				'name'     => 'body_class',
+				'priority' => 10,
+				'function' => 'filter_body_classes',
+			),
+			array(
+				'type'     => 'action',
+				'name'     => 'wp_head',
+				'priority' => 10,
+				'function' => 'add_pingback_link',
+			),
+			array(
+				'type'     => 'action',
+				'name'     => 'after_setup_theme',
+				'priority' => 10,
+				'function' => 'setup_theme',
+			),
+			array(
+				'type'     => 'action',
+				'name'     => 'init',
+				'priority' => 10,
+				'function' => 'add_title_tag_support',
+			),
+		);
+
+		// Check if hooks loaded.
+		foreach ( $hooks as $hook ) {
+
+			$this->assertEquals(
+				$hook['priority'],
+				call_user_func(
+					sprintf( 'has_%s', $hook['type'] ),
+					$hook['name'],
+					array(
+						$this->instance,
+						$hook['function'],
+					)
+				),
+				sprintf( 'BLANK_THEME::__construct() failed to register %1$s "%2$s" to %3$s()', $hook['type'], $hook['name'], $hook['function'] )
+			);
+		}
+		$this->assertEquals( 10, has_filter( 'excerpt_more', array( $this->instance, 'add_read_more_link' ) ) );
+		$this->assertEquals( 10, has_filter( 'body_class', array( $this->instance, 'filter_body_classes' ) ) );
+		$this->assertEquals( 10, has_action( 'wp_head', array( $this->instance, 'add_pingback_link' ) ) );
```

Comment:
are these duplicate test cases? does looping through `$hooks` cover this test cases?